File: app/src/main/java/com/example/megabest/Adapters/SimilarMoviesRecyclerView.java
package com.example.megabest.Adapters;

import android.content.Context;
import android.content.Intent;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.megabest.DetailsActivity;
import com.example.megabest.Model.Movie;
import com.example.megabest.R;
import com.makeramen.roundedimageview.RoundedImageView;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

import static android.content.ContentValues.TAG;

public class SimilarMoviesRecyclerView extends RecyclerView.Adapter<SimilarMoviesRecyclerView.SimilarMoviesViewHolder> {
    private Context context ;
    private List<Movie> movieList = new ArrayList<>();

    public SimilarMoviesRecyclerView(Context context, List<Movie> movieList) {
        this.context = context;
        this.movieList = movieList;
    }

    public Context getContext() {
        return context;
    }

    public void setContext(Context context) {
        this.context = context;
    }

    public List<Movie> getMovieList() {
        return movieList;
    }

    public void setMovieList(List<Movie> movieList) {
        this.movieList = movieList;
        notifyDataSetChanged();
    }

    @NonNull
    @NotNull
    @Override
    public SimilarMoviesViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.similar_movies,parent,false);
        return new SimilarMoviesRecyclerView.SimilarMoviesViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull @NotNull SimilarMoviesRecyclerView.SimilarMoviesViewHolder holder, int position) {
        holder.onBindView(holder, position);
    }

    @Override
    public int getItemCount() {
        return movieList.size();

    }

    public class SimilarMoviesViewHolder extends RecyclerView.ViewHolder {
        RoundedImageView similarMovieImage ;
        TextView similarMovieTitle ;
        Intent detailedActivity;
        public SimilarMoviesViewHolder(@NonNull @NotNull View itemView) {
            super(itemView);
            similarMovieImage=itemView.findViewById(R.id.similarMovieImage);
            similarMovieTitle=itemView.findViewById(R.id.similarMovieTitle);
        }
        public void onBindView(SimilarMoviesRecyclerView.SimilarMoviesViewHolder holder , int position){
            holder.similarMovieTitle.setText(movieList.get(position).getTitle().toString());
            Glide.with(holder.itemView.getContext())
                    .load("https://image.tmdb.org/t/p/w500/"+movieList.get(position).getPosterPath())
                    .into(holder.similarMovieImage);
            holder.similarMovieImage.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    detailedActivity =  new Intent(context, DetailsActivity.class);
                    detailedActivity.putExtra("BackPath Poster",movieList.get(position).getBackdrop_path());
                    detailedActivity.putExtra("OverView",movieList.get(position).getOverview());
                    detailedActivity.putExtra("Movie Title",movieList.get(position).getTitle());
                    detailedActivity.putExtra("Movie Id",String.valueOf(movieList.get(position).getId()));
                    Log.d(TAG, "onClick: movie id "+movieList.get(position).getId());
                    context.startActivity(detailedActivity);
                }
            });

                }
            }


}

File: app/src/main/java/com/example/megabest/app/Adapters/SimilarMoviesRecyclerView.kt
package com.example.megabest.app.Adapters

import android.content.ContentValues
import android.content.Context
import android.content.Intent
import android.transition.Fade
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.AnimationUtils
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.megabest.R
import com.example.megabest.app.Adapters.SimilarMoviesRecyclerView.SimilarMoviesViewHolder
import com.example.megabest.app.DetailsActivity
import com.example.megabest.entities.Movie
import com.makeramen.roundedimageview.RoundedImageView
import java.util.*

class SimilarMoviesRecyclerView(var context: Context, movieList: List<Movie>) :
    RecyclerView.Adapter<SimilarMoviesViewHolder>() {
    private var movieList: List<Movie> = ArrayList()
    fun getMovieList(): List<Movie> {
        return movieList
    }

    fun setMovieList(movieList: List<Movie>) {
        this.movieList = movieList
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SimilarMoviesViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.similar_movies, parent, false)
        return SimilarMoviesViewHolder(view)
    }

    override fun onBindViewHolder(holder: SimilarMoviesViewHolder, position: Int) {
        holder.onBindView(holder, position)
    }

    override fun getItemCount(): Int {
        return movieList.size
    }

    inner class SimilarMoviesViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var similarMovieImage: RoundedImageView
        var similarMovieTitle: TextView
        var detailedActivity: Intent? = null
        var similarMoviesCard: CardView
        fun onBindView(holder: SimilarMoviesViewHolder, position: Int) {
            holder.similarMoviesCard.animation = AnimationUtils.loadAnimation(
                context, R.anim.fade_transition
            )
            holder.similarMovieTitle.text = movieList[position].title.toString()
            Glide.with(holder.itemView.context)
                .load("https://image.tmdb.org/t/p/w500/" + movieList[position].posterPath)
                .into(holder.similarMovieImage)
            holder.similarMovieImage.setOnClickListener {
                detailedActivity = Intent(context, DetailsActivity::class.java)
                detailedActivity!!.putExtra("BackPath Poster", movieList[position].backdrop_path)
                detailedActivity!!.putExtra("OverView", movieList[position].overview)
                detailedActivity!!.putExtra("Movie Title", movieList[position].title)
                detailedActivity!!.putExtra("Movie Id", movieList[position].id.toString())
                Log.d(ContentValues.TAG, "onClick: movie id " + movieList[position].id)
                val fade = Fade()

                // ActivityOptions optionsCompat = ActivityOptions.makeSceneTransitionAnimation((Activity) context,similarMovieImage, ViewCompat.getTransitionName(similarMovieImage));
                context.startActivity(detailedActivity)
            }
        }

        init {
            similarMovieImage = itemView.findViewById(R.id.similarMovieImage)
            similarMovieTitle = itemView.findViewById(R.id.similarMovieTitle)
            similarMoviesCard = itemView.findViewById(R.id.similarMoviesCard)
        }
    }

    init {
        this.movieList = movieList
    }
}